### PR TITLE
Only analyse code on main workflow

### PR DESCRIPTION
**Pull request by Shinigami92**
## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

I get spammed by emails :laughing:  
I think/hope this performs only one code-analyse per push instead of two

### CODE REVIEW

**Comment by Shinigami92** [comment]
Seems to work: https://github.com/pioardi/poolifier/pull/208/checks?check_run_id=1942541835
Only node-version `'12.x'` performs the `Sonar Code Analysis` step

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
           CI: true
 
       - name: Sonar Code Analysis
-        if: "github.repository == 'pioardi/poolifier'"
+        if: "github.repository == 'pioardi/poolifier' && matrix.node-version == '12.x'"
```
Comment (jerome-benoit) [inline .github/workflows/ci.yml:50]: I would prefer to have it run on the current node.js LTS, not the previous one.

Comment (Shinigami92) [reply .github/workflows/ci.yml:50]: We are also releasing with node 12

https://github.com/pioardi/poolifier/blob/5716dd56de3de46c5871a6af3b9f02da897bc2e4/.github/workflows/npmpublish.yml#L47

@pioardi What is your opinion about that?

I'm not biased at all and don't care which version is used in the end :slightly_smiling_face:

Comment (pioardi) [reply .github/workflows/ci.yml:50]: I would prefer to support node 14.x.x better :)  In this case I don't think that really matter a lot, but please use the 14 version.

Comment (Shinigami92) [reply .github/workflows/ci.yml:50]:
```suggestion
        if: "github.repository == 'pioardi/poolifier' && matrix.node-version == '14.x'"
```

**Review by pioardi — APPROVED** [review]
Thanks @Shinigami92  this help to speed up our activities.
Please guys take a look into the chat @jerome-benoit great news !!!

**Pull request merged**